lib/Sema/CSSimplify.cpp

```diff
@@ -3399,6 +3399,13 @@ bool ConstraintSystem::repairFailures(
     break;
   }
 
+  case ConstraintLocator::InstanceType: {
+    if (lhs->hasUnresolvedType() || rhs->hasUnresolvedType())
```

Comment:
Instead of checking unresolved type directly should be actually frame it in terms of a hole e.g. `->hasHole()`?

Reply:
I can add `isHole` that does the same thing as `hasUnresolvedType`. I considered that for the last PR, but didn't know if it was worth having two methods that do the same thing. I do think we should rename it at least once CSDiag is gone, as the only use for `UnresolvedType` will be for holes. What do you think?

Reply:
I think we better introduce `hasHole` sooner than later because that gives up a clear distinction here. We can then remove `hasUnresolvedType` completely when the time comes.

Reply:
Though, if I do add `hasHole` now, I'd prefer to do that in my refactoring PR, so that this specific PR isn't renaming a bunch of stuff

Reply:
Sure, no problem!